### PR TITLE
Change: add support for filtered data in Pivot table visualization

### PR DESCRIPTION
**Pull request by deecay**
This PR changes how the `data` fed to pivottable visualization is prepared, and makes pivottable show filtered and formatted data just as you can see in Table visualization.

I have noticed that pivottable visualization:
* does not incorporate query/dashboard filter
* `::filter` annotation appears as part of pivottable attribute name and does not get removed
* date and datetime columns does not respect `REDASH_DATE_FORMAT`

This PR addresses all three points.

### CODE REVIEW

File: client/app/visualizations/pivot/index.js [hunk 1]
```diff
@@ -21,7 +22,19 @@ function pivotTableRenderer() {
...
           // it which interferes with other visualizations.
-          data = angular.copy($scope.queryResult.getRawData());
+          data = [];
+          const columns = $scope.queryResult.getColumns();
+          const rows = $scope.queryResult.getData();
+          rows.forEach((row) => {
+            const rowObj = {};
+            columns.forEach((col) => {
+              const colName = getColumnCleanName(col.name);
+              const value = formatValue($filter, clientConfig, row[col.name], col.type);
```
Comment (arikfr) [inline client/app/visualizations/pivot/index.js]: Two things that come to mind here:

1. Are we sure that converting numbers into strings will behave properly with PivotTable?
2. For dates, shouldn't we use some standard like (YYYY-MM-DD) regardless of configuration to solve #216?

Comment (deecay) [inline client/app/visualizations/pivot/index.js]: - Comment 2

  I am thinking of adding a Sorting option to pivottable editor. Because passing a sort-friendly format will work for numerical dates, but it doesn't help for Monday/Tuesday/..., Mon/Tue/..., January/February/..., Jan/Feb/..., etc.
 If we pass `REDAH_DATE_FORMAT` to pivot, the problem below will go away because the dates won't be in ISO format,



and letting people define custom sorters will take care of the first comment of #216.

- Comment 1
I have to look into it a bit more for part 1 of your comment.

Comment (arikfr) [inline client/app/visualizations/pivot/index.js]: They have a way to pass a sort function?

Anyway, how about we change this pull request to only handle the filters part and leave the formatting to another pull request?

Comment (deecay) [inline client/app/visualizations/pivot/index.js]: Yes they do. Example is [here](http://nicolas.kruchten.com/pivottable/examples/mps_agg.html).
I have to come up with how to let people define it in pivottable editor, but currently have no idea.

I'm perfectly fine with dividing this PR into two, and making this only about the filters.
I assume what you mean is just replacing `.getRawData()` to `.getData()`?

**Comment by deecay** [comment]
This PR has been changed and now it is about fixing pivottable of filtered data.

Formatting part will be addressed in a separate PR.

File: client/app/visualizations/pivot/index.js (hunk 1 above)
Comment (arikfr) [inline client/app/visualizations/pivot/index.js]: > I have to come up with how to let people define it in pivottable editor, but currently have no idea.

The user shouldn't have to define them. We should have default ones for the types we have (dates, numbers, etc - where needed).
